src/OnboardingSPA/components/GlobalStyleParent/index.js

```diff
@@ -0,0 +1,74 @@
+import { useSelect, useDispatch } from '@wordpress/data'; 
+import { useState, useEffect } from '@wordpress/element'; 
+
+import { store as nfdOnboardingStore } from '../../store';
+import { getGlobalStyles, setGlobalStyles } from '../../utils/api/themes';
+import { useGlobalStylesOutput } from '../../utils/global-styles/use-global-styles-output';
+
+/**
+ * Global Style Parent Component
+ * The Fetching of Global Style Object from either store or API is 
+ * common to a lot many places and this component does the trick import { useState, useEffect } from '@wordpress/element';for us.
+ *
+ * @returns Global Style Parent
+ */
+
+const GlobalStyleParent = ({ children }) => {
+
+    const [isLoaded, setIsLoaded] = useState(false);
+
+    const {
+        currentData,
+        storedPreviewSettings,
+    } = useSelect((select) => {
+        return {
+            currentData:
+                select(nfdOnboardingStore).getCurrentOnboardingData(),
+            storedPreviewSettings:
+                select(nfdOnboardingStore).getPreviewSettings(),
+        };
+    }, []);
+
+    const { updateThemeStatus, updatePreviewSettings } =
+        useDispatch(nfdOnboardingStore);
+
+    const getStylesAndPatterns = async () => {
+        const globalStyles = await getGlobalStyles();
+        if (globalStyles?.error) {
+            return updateThemeStatus(THEME_STATUS_NOT_ACTIVE);
+        }
+        let selectedGlobalStyle;
+        if (storedPreviewSettings?.title)
```

Comment:
Not a good idea to rely on `title` better to check if it is an `object` that has `settings` and `styles`

Reply:
Sure but i found that styles and GlobalStyles were used so can't rely on them too, sure will add the settings check too.